src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_LENGTH = 10;
+
+    private int size;
+    private K[] keys = (K[]) new Object[MAX_ARRAY_LENGTH];
+    private V[] values = (V[]) new Object[MAX_ARRAY_LENGTH];
+
     @Override
     public void put(K key, V value) {
+        int checkedIndex = checkIndex(key);
+        if (checkedIndex >= 0) {
+            values[checkedIndex] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
+    }
+
+    private int checkIndex(K key) {
```

Comment:
If method's name starts with "check" - it is expected that it will returns boolean value. It's more logical to use "get" here
```suggestion
    private int getIndex(K key) {
```